pycytominer/cyto_utils/DeepProfiler_processing.py

```diff
@@ -0,0 +1,280 @@
+
+        the file_aggregate dictionary contains the file locations and metadata for each grouping.
+        If for example we are grouping by well then the keys of self.file_aggregate would be:
+        plate1/well1, plate1/well2, plate2/well1, etc.
+        """
+        if not hasattr(self, "filenames"):
+            self.build_filenames()
+
+        self.file_aggregate = {}
+        for filename in self.filenames:
+            file_info = self.extract_filename_metadata(
+                filename, self.filename_delimiter
+            )
+            file_key = file_info[self.aggregate_on]
+
+            if self.aggregate_on == "site":
+                file_key = (
+                    f"{file_info['plate']}/{file_info['well']}_{file_info['site']}"
+                )
+
+            if self.aggregate_on == "well":
+                file_key = f"{file_info['plate']}/{file_info['well']}"
+
+            if file_key in self.file_aggregate:
+                self.file_aggregate[file_key]["files"].append(filename)
+            else:
+                self.file_aggregate[file_key] = {}
+                self.file_aggregate[file_key]["files"] = [filename]
+
+            self.file_aggregate[file_key]["metadata"] = file_info
+
+    def aggregate_deep(self):
+        """
+        Aggregates the profiles into a pandas dataframe.
+
+        For each key in file_aggregate, the profiles are loaded, concatenated and then aggregated.
+        If files are missing, we throw a warning but continue the code.
+        """
+        if not hasattr(self, "file_aggregate"):
+            self.setup_aggregate()
+
+        self.aggregated_profiles = []
+        self.aggregate_merge_col = f"Metadata_{self.aggregate_on.capitalize()}_Position"
+
+        for metadata_level in self.file_aggregate:
+            # uses custom load function to create df with metadata and profiles
+            arr = [load_npz(x) for x in self.file_aggregate[metadata_level]["files"]]
+            # empty dataframes from missing files are deleted
+            arr = [x for x in arr if not x.empty]
+            # if no files were found there is a miss-match between the index and the output files
+            if not len(arr):
+                warnings.warn(
+                    f"No files for the key {metadata_level} could be found.\nThis program will continue, but be aware that this might induce errors!"
+                )
+                continue
+
+            df = pd.concat(arr)
+
+            # Prepare inputs for the aggregate function
+            meta_df = pd.DataFrame(
+                self.file_aggregate[metadata_level]["metadata"], index=[0]
+            ).reset_index(drop=True)
+            meta_df.columns = [
+                f"Metadata_{x.capitalize()}" if not x.startswith("Metadata_") else x
+                for x in meta_df.columns
+            ]
+
+            # site values are int
+            meta_df.Metadata_Site = meta_df.Metadata_Site.astype(int)
```

Comment:
I am a bit concerned about this line. The function will error out if the info in Metadata_Site cannot be coerced to int. Is this line necessary?

Reply:
Yea, its part of the spageti code of this function. 

the index files has the sites and ints and I think they should be ints. 
But when you read them from the npz files, then you get str... so one of them has to be changed

Reply:
but why do you need them to be ints? You can groupby str variables and I think that's all we need this info for (aggregation)

Reply:
In 353cb382db3783a9febaca8fc02c96598f2bebfd I made sure that all metadata columns have an `str` dtype upon instantiation.

I think this is our safest bet.

Reply:
OK sure, checks out now